api-notification.go

```diff
@@ -156,6 +157,14 @@ func (c Client) ListenBucketNotification(bucketName string, accountArn Arn, done
 			return
 		}
 
+		// Check if we are calling Minio server which supports listening to bucket notifications
+		if !strings.HasPrefix(resp.Header.Get("Server"), "Minio/") {
```

Comment:
> This RFC is deprecated can you check the latest?

yes, you wer right, in recent specs, the paragraph about not touching the server header is removed (compare this https://www.w3.org/Protocols/rfc2616/rfc2616-sec14.html#sec14.38 and https://tools.ietf.org/html/rfc7231#page-73)

In addition, I noticed that some reverse proxies put their own names in the server field and some will remove it.
https://trac.nginx.org/nginx/ticket/538
http://cherokee-project.com/doc/modules_handlers_proxy.html
https://caddyserver.com/docs/header
https://twitter.com/caddyserver/status/714587023948931072

Clearly, users are not reluctant to remove the Server header in their proxies, so better to avoid this solution.